### PR TITLE
Let pygmt.info load datetime columns into a str dtype array

### PR DESCRIPTION
**Pull request by weiji14**
**Description of proposed changes**

Fixes problem with pygmt.info not being able to handle datetime64 inputs. I.e. `ValueError: could not convert string to float: '2021-01-01T12:34:56'`.

However, users will still need to use`pygmt.info(..., f="0T")` until upstream issue at https://github.com/GenericMappingTools/gmt/issues/4241 is resolved. Something like so:

```python
import pandas as pd
import pygmt

table = pd.date_range(start="2020-01-01", periods=5).to_numpy()
output = pygmt.info(table=table, per_column=True, f="0T")
print(output)
# ['2020-01-01T00:00:00' '2020-01-05T00:00:00']
```

Also added two extra unit tests using numpy datetime64 types.

References:
- https://stackoverflow.com/questions/31553983/python-using-numpy-loadtxt-reading-a-csv-file-with-different-data-types-for-ea/31554777#31554777

Addresses #597 on the pygmt side using fix mentioned in https://github.com/GenericMappingTools/pygmt/issues/597#issuecomment-725091467. Assuming that https://github.com/GenericMappingTools/gmt/issues/4241 is fixed in GMT 6.2.0, then issue #597 can be closed once PyGMT bumps minimum required GMT version to 6.2.0.

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/tests/test_info.py [hunk 1]
```diff
@@ -41,6 +41,18 @@ def test_info_dataframe():
     assert output == expected_output
 
 
+def test_info_numpy_array_time_column():
+    """
+    Make sure info works on a numpy.ndarray input with a datetime type.
+    """
+    table = pd.date_range(start="2020-01-01", periods=5).to_numpy()
+    output = info(table=table, f="0T")
```
Comment (weiji14) [inline pygmt/tests/test_info.py:49]: The `f="0T"` argument can be removed once upstream GMT fixes https://github.com/GenericMappingTools/gmt/issues/4241.

File: pygmt/tests/test_info.py [hunk 2]
```diff
@@ -115,6 +127,17 @@ def test_info_per_column():
     )
 
 
+def test_info_per_column_with_time_inputs():
+    """
+    Make sure the per_column option works with time inputs.
+    """
+    table = pd.date_range(start="2020-01-01", periods=5).to_numpy()
+    output = info(table=table, per_column=True, f="0T")
```
Comment (weiji14) [inline pygmt/tests/test_info.py:135]: Actually, do we want a proper long alias for [`f`](https://docs.generic-mapping-tools.org/6.1/gmt.html#f-full) here?

File: pygmt/src/info.py [hunk 3]
```diff
@@ -94,6 +94,10 @@ def info(table, **kwargs):
             # instead of a raw string that is less useful.
             if result.startswith(("-R", "-T")):  # e.g. -R0/1/2/3 or -T0/9/1
                 result = result[2:].replace("/", " ")
-            result = np.loadtxt(result.splitlines())
+            try:
+                result = np.loadtxt(result.splitlines())
+            except ValueError:
+                # Load non-numerical outputs in str type, e.g. for datetime
+                result = np.loadtxt(result.splitlines(), dtype="str")
```
Comment (seisman) [inline pygmt/src/info.py:101]: It means all the output columns will be in str type, including numerical columns. I'm not sure if I like it.

Comment (weiji14) [reply pygmt/src/info.py:101]: Yes, but there's no way to have mixed types in numpy. An array with str (e.g. datetime) and numerical (e.g. int/float) types will have to be made into a `str` or `object` type.

File: pygmt/tests/test_info.py (hunk 2 above)
Comment (seisman) [reply pygmt/tests/test_info.py:135]: Yes.

File: pygmt/src/info.py (hunk 3 above)
Comment (seisman) [reply pygmt/src/info.py:101]: I'm OK with a string-type array. We can think more about the return values later (https://github.com/GenericMappingTools/pygmt/issues/593).

File: pygmt/tests/test_info.py (hunk 2 above)
Comment (weiji14) [reply pygmt/tests/test_info.py:135]: Ok, will do in a separate PR.

**Comment by weiji14** [comment]
Hmm, that crash on the Windows GMT dev test at https://github.com/GenericMappingTools/pygmt/runs/2025836161?check_suite_focus=true#step:15:86 at the start (before any tests got run) doesn't look nice.

**Comment by seisman** [comment]
/test-gmt-dev

> Hmm, that crash on the Windows GMT dev test at [GenericMappingTools/pygmt/runs/2025836161?check_suite_focus=true#step:15:86](https://github.com/GenericMappingTools/pygmt/runs/2025836161?check_suite_focus=true#step:15:86) at the start (before any tests got run) doesn't look nice.

It may be just a sporadic crash. Just re-triggered the dev tests.

**Comment by weiji14** [comment]
Ok, Windows tests have passed at https://github.com/GenericMappingTools/pygmt/runs/2026019634?check_suite_focus=true#step:15:867. Any other comment on this PR?

**Review by seisman — APPROVED** [review]

**Pull request merged**